x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -242,26 +242,35 @@ export class CsvGenerator {
   /*
    * Format a Datatable into rows of CSV content
    */
-  private generateRows(
+  private async generateRows(
     columns: string[],
     table: Datatable,
     builder: MaxSizeStringBuilder,
     formatters: Record<string, FieldFormat>,
     settings: CsvExportSettings
   ) {
     this.logger.debug(`Building ${table.rows.length} CSV data rows...`);
+
+    const asyncGenerateRow = async (dataTableRow: Record<string, any>): Promise<string> => {
+      return new Promise((resolve) => {
+        setImmediate(() => {
+          resolve(
+            columns
+              .map((f) => ({ column: f, data: dataTableRow[f] }))
+              .map(this.formatCellValues(formatters))
+              .map(this.escapeValues(settings))
+              .join(settings.separator) + '\n'
+          );
+        });
+      });
+    };
+
     for (const dataTableRow of table.rows) {
       if (this.cancellationToken.isCancelled()) {
         break;
       }
 
-      const row =
-        columns
-          .map((f) => ({ column: f, data: dataTableRow[f] }))
-          .map(this.formatCellValues(formatters))
-          .map(this.escapeValues(settings))
-          .join(settings.separator) + '\n';
-
+      const row = await asyncGenerateRow(dataTableRow);
```

Comment:
The following code should do the same but without introducing unnecessary callback.

```suggestion
      await new Promise(setImmediate);
      
      const row =
        columns
          .map((f) => ({ column: f, data: dataTableRow[f] }))
          .map(this.formatCellValues(formatters))
          .map(this.escapeValues(settings))
          .join(settings.separator) + '\n';
```

Reply:
https://github.com/elastic/kibana/pull/120309/commits/ebacdbbb0640cc0b6577475dbc9be3b814b8d99d